### PR TITLE
Use pprof to get flamegraph of get_page and get_relsize requests.

### PR DESCRIPTION
**Pull request by hlinnaka**
This depends on a hacked version of the 'pprof-rs' crate. Because of
that, it's put under an optional 'profling' feature. It is enabled by
default though.

The flamegraph is written to 'flamegraph.svg' in the pageserver
workdir when the 'pageserver' process exits.

Add a performance test that runs the perf_pgbench test, with profiling
enabled.

### CODE REVIEW

**Comment by hlinnaka** [comment]
The 'pprof-rs' crate changes could use some cleanup, but let's see if we can get a flamegraph from the CI now.

**Comment by hlinnaka** [comment]
The flamegraph can be seen in the Artifacts of the 'benchmark' Circle CI job. For example [here](https://output.circle-artifacts.com/output/job/5ef41975-7894-4dbb-a548-8b340f3c9637/artifacts/0/tmp/test_output/test_pgbench_flamegraph/repo/flamegraph.svg):

File: zenith_utils/src/pq_proto.rs [hunk 1]
```diff
@@ -100,6 +100,21 @@ pub struct FeExecuteMessage {
 #[derive(Debug)]
 pub struct FeCloseMessage {}
 
+/// Retry a read on EINTR
+///
+/// This runs the enclosed expression, and if it returns
+/// Err(io::ErrorKind::Interrupted), retries it.
+macro_rules! retry_read {
```
Comment (knizhnik) [inline zenith_utils/src/pq_proto.rs:107]: Why it is necessary to define `retry_read` as macro, not as a function?

**Review by knizhnik — APPROVED** [review]

File: zenith_utils/src/pq_proto.rs (hunk 1 above)
Comment (hlinnaka) [reply zenith_utils/src/pq_proto.rs:107]: I tried that first. Maybe it'd be possible, but I couldn't make it work.

**Pull request merged**